src/controllers/book.js

```diff
@@ -0,0 +1,119 @@
+const db = require("../models");
+const Book = db.book;
+const Op = db.Sequelize.Op;
+
+exports.create = (req, res) => {
+  // Validate request
+  if (!req.body.title || !req.body.author) {
+    res.status(400).send({
+      message: "Title and Author are required!"
+    });
+    return;
+  }
+
+  let pubDate = new Date().toString();
+  if (req.body.publicationDate) {
+    pubDate = req.body.publicationDate.toString();
+  }
+
+  // Create a book
+  const book = {
+    title: req.body.title,
+    description: req.body.description,
+    author: req.body.author,
+    publicationDate: pubDate,
+    resume: req.body.abstract,
+    cover: req.body.cover
+  };
+
+  // Save Book in the database
+  Book.create(book)
+    .then(data => {
+      res.send(data);
+    })
+    .catch(err => {
+      res.status(500).send({
+        message:
+          err.message || "Some errors ocurred while creating the book"
+      });
+    });
+};
+
+exports.findAll = (req, res) => {
+  const title = req.query.title;
+  var condition = title ? { title: { [Op.iLike]: `%${title}%` } } : null;
+
+  Book.findAll({ where: condition })
+    .then(data => {
+      res.send(data);
+    })
+    .catch(err => {
+      res.status(500).send({
+        message:
+          err.message || "Some error occurred while searching your books."
+      });
+    });
+};
+
+exports.findOne = (req, res) => {
+  const id = req.params.id;
+
+  Book.findByPk(id)
+    .then(data => {
```

Comment:
What will happen if the id in params does not exist?

Reply:
that case would never happen xd,